iosApp/iosApp/Pages/Map/HomeMapViewHandlerExtension.swift

```diff
@@ -219,15 +219,15 @@ extension HomeMapView {
         guard let tripId = vehicle.tripId else { return }
         guard case .stopDetails = nearbyVM.navigationStack.lastSafe() else { return }
 
-        let departures = nearbyVM.departures
-        let patterns = departures?.routes.first(where: { patterns in
-            patterns.routes.contains { $0.id == vehicle.routeId }
-        })
-        let trip = patterns?.allUpcomingTrips().first(where: { upcoming in
+        let routeCardData = nearbyVM.routeCardData
+        let route = routeCardData?.first(where: { $0.lineOrRoute.containsRoute(routeId: vehicle.routeId) })
+        let stop = route?.stopData.first
+        let allTrips = stop?.data.flatMap(\.upcomingTrips)
+        let trip = allTrips?.first(where: { upcoming in
             upcoming.trip.id == tripId
         })
         let stopSequence = trip?.stopSequence
-        let routeId = trip?.trip.routeId ?? vehicle.routeId ?? patterns?.routeIdentifier
+        let routeId = trip?.trip.routeId ?? vehicle.routeId ?? route?.lineOrRoute.id
```

Comment:
👍 I was worried at first that using `lineOrRoute.id` would cause issues on the GL, but see that `patterns?.routeIdentifier` would also return the line id.